src/lib/profiles/data-management/Current/TraitCatalog.h

```diff
@@ -194,296 +194,6 @@ class SingleResourceTraitCatalog : public TraitCatalogBase<T>
 typedef SingleResourceTraitCatalog<TraitDataSink> SingleResourceSinkTraitCatalog;
 typedef SingleResourceTraitCatalog<TraitDataSource> SingleResourceSourceTraitCatalog;
 
```

Comment:
Might want to place extern template instantiations here to make sure the compiler generate doesn't any code for these types when TraitCatalog.h is included.  E.g.:

```
extern template class SingleResourceTraitCatalog<TraitDataSink>;
extern template class SingleResourceTraitCatalog<TraitDataSource>;
```

Reply:
It would a bit of a belt-and-suspenders approach here? The way the template is split up right now we should be achieving the same behavior.  The above approach would ensure that if `SingleResourceTraitCatalog.h`was subsequently altered to include definitions of actual functions/linking symbols, those would not be compiled here, but we already stipulate such behavior via `.h`/`.ipp` split.

Reply:
I think I will merge this as is -- I find it hard to think about the interactions of `typedef`, template instantiation, and `extern`.  We can clean up in a subsequent PR